pkg/controllers/securitypolicy/securitypolicy_controller.go

```diff
@@ -163,7 +163,7 @@ func (r *SecurityPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	} else {
 		if controllerutil.ContainsFinalizer(obj, servicecommon.SecurityPolicyFinalizerName) {
 			metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerDeleteTotal, MetricResType)
-			if err := r.Service.DeleteSecurityPolicy(obj, false, servicecommon.ResourceTypeSecurityPolicy); err != nil {
+			if err := r.Service.DeleteSecurityPolicy(obj.UID, false, servicecommon.ResourceTypeSecurityPolicy); err != nil {
```

Comment:
Why here changes `obj` to `obj.UID`? I have this question because in `networkpolicy_controller.go` we also have the similar usage `r.Service.DeleteSecurityPolicy(networkPolicy, false, servicecommon.ResourceTypeNetworkPolicy)`

Reply:
good point. In the very beginning of SP implementation, we use obj.UID to delete securitypolicy, the purpose of passing UID instead is trying to get SP from store not recreate it in deleteSecurityPolicy function for the
case *v1alpha1.SecurityPolicy: 

Later, I changed it to obj from obj.UID, because I want to leverage the processing buildSecurityPolicy to handle all situation for project related resources. 
case *v1alpha1.SecurityPolicy:
nsxSecurityPolicy, nsxGroups, projectShares, err = service.buildSecurityPolicy(sp, createdFor)

By now, I realized that we don't need passing obj anymore since we we have project store to save project related resource and send one API call. the whole process is simplified. 

But for r.Service.DeleteSecurityPolicy(networkPolicy, false, servicecommon.ResourceTypeNetworkPolicy)

I also checked the code,and I found that we might not be able to pass  networkPolicy.UID in DeleteSecurityPolicy because networkpolicy will call convertNetworkPolicyToInternalSecurityPolicies in DeleteSecurityPolicy to do conversion:
case *networkingv1.NetworkPolicy:
		internalSecurityPolicies, err := service.convertNetworkPolicyToInternalSecurityPolicies(obj.(*networkingv1.NetworkPolicy))

Reply:
I'd like to listen your advice.

Reply:
> By now, I realized that we don't need passing obj anymore since we we have project store to save project related resource and send one API call. the whole process is simplified.

Is this change necessary to the topic `Change securitypolicy cleanup to get VPC info by NSX path`? If not, I can try to unify the whole process for both sp and np in another patch and we can keep your change in this patch.

Reply:
sounds good,  it's good that we can rethink to try to unify the both SP and NP deletion. I can remove this change since it's not absolute necessary for this patch.

Reply:
lgtm